contracts/contribute/ContributeManager.sol

```diff
@@ -314,17 +373,114 @@ contract ContributeManager is ERC721TokenReceiver {
             revert ServiceNotDefined(socialId);
         }
 
+        _unstake(msg.sender, socialId, serviceId, multisig, stakingInstance, pullService);
+
+        _locked = 1;
+    }
+
+    /// @dev Unstakes service Id corresponding to the msg.sender and clears the contributor record.
+    function _unstake(
+        address contributor,
+        uint256 socialId,
+        uint256 serviceId,
+        address multisig,
+        address stakingInstance,
+        bool pullService
+    ) internal {
         // Unstake the service
         IStaking(stakingInstance).unstake(serviceId);
 
-        // Transfer the service back to the original owner
-        INFToken(serviceRegistry).transferFrom(address(this), msg.sender, serviceId);
+        // Terminate service
+        (, uint256 terminateRefund) = IService(serviceManager).terminate(serviceId);
+
+        // Unbond service
+        (, uint256 unbondRefund) = IService(serviceManager).unbond(serviceId);
+
+        // Calculate refund
+        uint256 refund = terminateRefund + unbondRefund;
+
+        // Transfer back OLAS tokens
+        IToken(olas).transfer(contributor, refund);
```

Comment:
This was the original missing part that let to the current situation of not being able to fully unbond the service.